src/thor/bidirectional_astar.cc

```diff
@@ -658,13 +660,14 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
       } else {
         // Search is exhausted. If a connection has been found, return it
         if (best_connections_.empty()) {
-          // No route found.
-          LOG_ERROR("Bi-directional route failure - forward search exhausted: n = " +
-                    std::to_string(edgelabels_forward_.size()) + "," +
-                    std::to_string(edgelabels_reverse_.size()));
-          return {};
+          // mark the forward direction exhausted so we dont expand from here
+          // anymore
+          forward_exhausted = true;
+          LOG_WARN("Forward search exhausted: n = " + std::to_string(edgelabels_forward_.size()) +
+                   "," + std::to_string(edgelabels_reverse_.size()));
+        } else {
```

Comment:
this is a nitpick but i feel like at this point we should flip the logic around to be like:

```c++
if (!best_connections_empty() {
  return FormPath(graphreader, options, origin, destination, forward_time_info, invariant);
}
// mark the forward direction exhausted so we dont expand from here anymore
forward_exhausted = true;
LOG_WARN("Forward search exhausted: n = " + std::to_string(edgelabels_forward_.size()) + "," + std::to_string(edgelabels_reverse_.size()));
``

Reply:
will do